cicecore/cicedynB/analysis/ice_history.F90

```diff
@@ -565,33 +565,35 @@ subroutine init_hist (dt)
              "none", c1, c0,                                      &
              ns1, f_aice)
 
-         if (grid_system == 'CD') then
-           call define_hist_field(n_uvele,"uvele","m/s",estr2D, ecstr,  &
+         call define_hist_field(n_uvele,"uvele","m/s",estr2D, ecstr,  &
```

Comment:
So we are defining the new fields unconditionnally, irrespective of `grid_system`. I guess that's because the field definition themselves do not take that much memory, so we don't care ?

Reply:
The new fields should only be allocated with grid_system == 'CD'.  There is some logic in the history mechanism that prevents fields from being accumulated and/or written unless the f_ flag is true.  And these extra variables with be set to false unless grid_system='CD'.  So it's sort of hidden.